### ISSUE TITLE
Spear strap torso encumbrance should drop when you draw the spear, but it doesn't

### ISSUE THREAD

**Issue by morikal** [CleverRaven/Cataclysm-DDA#49978]
### Describe the bug

Spear strap has 14 encumbrance when my steel spear is in it, and 2 encumbrance when empty. But when I draw the spear, my torso encumbrance stayed the same (counting 14 from the strap).

### Steps To Reproduce

I was wearing chain mail, plate mail, and the spear strap (see screenshot). Torso encumbrance 34.
I wielded the spear, emptying the strap. Torso encumbrance stayed 34 (though if you look on the right side of screenshot 2, the encumbrances should add up to 22).

If I remove & re-wear the spear strap, the encumbrance is correct (20 when I take it off, 22 after I put it back on).

### Expected behavior

Encumbrance should reduce by 12 on the torso when the spear is wielded.

### Screenshots

![After wield spear](https://user-images.githubusercontent.com/2304755/125886649-c20f002a-d928-45cc-bdf2-bc887b9761aa.jpg)
![Starting state](https://user-images.githubusercontent.com/2304755/125886651-7692ac40-7115-476c-aa85-4e42df22bc7f.jpg)
![after removing and rewearing the spear strap](https://user-images.githubusercontent.com/2304755/125886652-73208beb-8650-451b-8b41-4a95a68c888e.jpg)

### Versions and configuration

- OS: Windows 10 x64
- Game Version: 7bd735e (experimental build 2021-07-14-0029)
- Graphics version: Tiles
- Ingame language: English
- Mods loaded: Magiclysm, No Fungal Monsters

**Comment by Qrox** [comment]
If you skip one turn after drawing the spear, does the encumberance change?

**Comment by morikal** [comment]
I don't believe it did, but this character died and I can no longer check. (95% sure I waited a few turns and it didn't change.)

**Comment by casswedson** [comment]
Can not reproduce with the following steps.
* Wear a spear strap with a steel spear inside it, encumbrance is 14.
* Activate the spear strap to wield the spear, or use the wield command to get the spear, encumbrance immediately drops to 2.

![spearinstrap](https://user-images.githubusercontent.com/58050969/126079418-87a7fcb2-12ab-4ed5-9c28-5c431d7fe819.png)

![spearout](https://user-images.githubusercontent.com/58050969/126079420-624b188a-9f79-46ee-baa8-bd4e4afb0875.png)

- OS: Linux
    - OS Version: LSB Version: n/a; Distributor ID: ManjaroLinux; Description: Manjaro Linux; Release: 21.1.0; Codename: Pahvo; 
- Game Version: 21ca15ef [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Bionic Professions [package_bionic_professions]
]

**Comment by Zeropol** [comment]
Tried to reproduce, the encumbrance displayed changed after drawing a spear _and_ either moving or skipping a turn with "5".
Same behavior when putting a spear in a spear strap ( had to wait a turn for the displayed number to change )

**Comment by hexagonrecursion** [comment]
Confirmed.
* Chain mail armor and plate armor are irrelevant - I can reproduce the bug without them.

> If you skip one turn after drawing the spear, does the encumbrance change?

The encumbrance **does** drop to the correct value if I skip one turn

- OS: Linux
    - OS Version: LSB Version: :core-4.1-amd64:core-4.1-noarch; Distributor ID: Fedora; Description: Fedora release 33 (Thirty Three); Release: 33; Codename: ThirtyThree; 
- Game Version: 8051d7d [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Bionic Professions [package_bionic_professions]
]
